extensions/azurecore/src/account-provider/utils/msalCachePlugin.ts

```diff
@@ -194,26 +201,42 @@ export class MsalCachePluginProvider {
 		}
 	}
 
-	private async readCache(config: CacheConfiguration): Promise<string> {
-		config.lockTaken = await this.waitAndLock(config.lockFilePath, config.lockTaken);
+	/**
+	 * Reads from an encrypted cache file based on currentConfig provided.
+	 * @param currentConfig Currently used cache configuration.
+	 * @param alternateConfig Alternate cache configuration for resetting needs.
+	 * @returns Decrypted data.
+	 */
+	private async readCache(currentConfig: CacheConfiguration, alternateConfig: CacheConfiguration): Promise<string> {
+		currentConfig.lockTaken = await this.waitAndLock(currentConfig.lockFilePath, currentConfig.lockTaken);
 		try {
-			const cache = await fsPromises.readFile(config.cacheFilePath, { encoding: 'utf8' });
+			const cache = await fsPromises.readFile(currentConfig.cacheFilePath, { encoding: 'utf8' });
 			const decryptedData = await this._fileEncryptionHelper.fileOpener(cache!, true);
 			return decryptedData;
 		} catch (e) {
 			if (e.code === 'ENOENT') {
 				// File doesn't exist, log and continue
-				Logger.verbose(`MsalCachePlugin: Cache file for '${config.name}' cache not found on disk: ${e.code}`);
+				Logger.verbose(`MsalCachePlugin: Cache file for '${currentConfig.name}' cache not found on disk: ${e.code}`);
 			}
 			else {
 				Logger.error(`MsalCachePlugin: Failed to read from cache file: ${e}`);
-				Logger.verbose(`MsalCachePlugin: Error occurred when trying to read cache file, file will be deleted: ${e.message}`);
-				await fsPromises.unlink(config.cacheFilePath);
+				Logger.verbose(`MsalCachePlugin: Error occurred when trying to read cache file ${currentConfig.name}, file will be deleted: ${e.message}`);
+				await fsPromises.unlink(currentConfig.cacheFilePath);
+
+				// Ensure both configurations are not same.
+				if (currentConfig.name !== alternateConfig.name) {
```

Comment:
Is it possible they have the same name but different filepaths?

Reply:
We don't have such configurations